core/unit_test/TestMDRangeReduce.hpp

```diff
@@ -0,0 +1,63 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+template <typename T>
+void MDRangeReduceTester(int bound, int k) {
```

Comment:
Lol, `icpc`
```
/__w/kokkos/kokkos/core/unit_test/TestMDRangeReduce.hpp(24): error #869: parameter "bound" was never referenced
  void MDRangeReduceTester(int bound, int k) {
                               ^
```

What the hell do we even do with that? Mark it `[[maybe_unused]]` and pretend that this compiler is not bonkers?